File: launchMode/src/main/java/com/example/lt/myapplication/MainActivity.java
package com.example.lt.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.e(TAG, "onCreate: =======");
    }

    public void onClick(View view) {
        Intent intent = new Intent(this, SecondActivity.class);
        startActivity(intent);
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.e(TAG, "onStart: ========");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.e(TAG, "onResume: ============");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.e(TAG, "onStop: ============");

    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.e(TAG, "onPause: ============");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.e(TAG, "onDestroy: ==========");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.e(TAG, "onRestart: ==========");
    }
}

File: kotlinstudy/src/main/java/com/example/kotlinstudy/MainActivity.kt
package com.example.kotlinstudy

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.kotlinstudy.callback.CallbackTestActivity
import com.example.kotlinstudy.extention.moveTo
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnCallback.setOnClickListener {
            this@MainActivity.moveTo(CallbackTestActivity::class.java)
        }
        btnCoroutinePerformance.setOnClickListener {
            //性能测试
            moveTo(TestCoroutinePerformanceActivity::class.java)
        }
    }

    override fun onResume() {
        super.onResume()
        Log.e("kotlin", "onResume")
    }
}